packages/features/bookings/lib/handleSeats/handleSeats.ts

```diff
@@ -0,0 +1,132 @@
+import { HttpError } from "@calcom/lib/http-error";
+import prisma from "@calcom/prisma";
+import { BookingStatus } from "@calcom/prisma/enums";
+
+import { createLoggerWithEventDetails } from "../handleNewBooking";
+import createNewSeat from "./create/createNewSeat";
+import rescheduleSeatedBooking from "./reschedule/rescheduleSeatedBooking";
+import type { NewSeatedBookingObject, SeatedBooking, HandleSeatsResultBooking } from "./types";
+
+const handleSeats = async (newSeatedBookingObject: NewSeatedBookingObject) => {
+  const {
+    eventType,
+    reqBodyUser,
+    rescheduleUid,
+    reqBookingUid,
+    invitee,
+    bookerEmail,
+    smsReminderNumber,
+    eventTypeInfo,
+    uid,
+    originalRescheduledBooking,
+    reqBodyMetadata,
+    eventTypeId,
+    subscriberOptions,
+    eventTrigger,
+  } = newSeatedBookingObject;
+  const { evt } = newSeatedBookingObject;
+  const loggerWithEventDetails = createLoggerWithEventDetails(eventType.id, reqBodyUser, eventType.slug);
+
+  let resultBooking: HandleSeatsResultBooking = null;
+
+  const seatedBooking: SeatedBooking | null = await prisma.booking.findFirst({
+    where: {
+      OR: [
+        {
+          uid: rescheduleUid || reqBookingUid,
+        },
+        {
+          eventTypeId: eventType.id,
+          startTime: evt.startTime,
+        },
+      ],
+      status: BookingStatus.ACCEPTED,
+    },
+    select: {
+      uid: true,
+      id: true,
+      attendees: { include: { bookingSeat: true } },
+      userId: true,
+      references: true,
+      startTime: true,
+      user: true,
+      status: true,
+      smsReminderNumber: true,
+      endTime: true,
+      scheduledJobs: true,
+    },
+  });
+
+  if (!seatedBooking) {
+    throw new HttpError({ statusCode: 404, message: "Could not find booking" });
+  }
+
+  // See if attendee is already signed up for timeslot
+  if (
+    seatedBooking.attendees.find((attendee) => attendee.email === invitee[0].email) &&
+    dayjs.utc(seatedBooking.startTime).format() === evt.startTime
+  ) {
+    throw new HttpError({ statusCode: 409, message: "Already signed up for this booking." });
+  }
+
+  // There are two paths here, reschedule a booking with seats and booking seats without reschedule
+  if (rescheduleUid) {
+    resultBooking = await rescheduleSeatedBooking(
+      // Assert that the rescheduleUid is defined
+      { ...newSeatedBookingObject, rescheduleUid },
+      seatedBooking,
+      resultBooking,
+      loggerWithEventDetails
+    );
+  } else {
+    resultBooking = await createNewSeat(newSeatedBookingObject, seatedBooking);
+  }
+
+  // Here we should handle every after action that needs to be done after booking creation
+
+  // Obtain event metadata that includes videoCallUrl
+  const metadata = evt.videoCallData?.url ? { videoCallUrl: evt.videoCallData.url } : undefined;
+  try {
+    await scheduleWorkflowReminders({
```

Comment:
same as for `handleWebhookTrigger` below, workflow is triggered twice on reschedule when booking seat